### PR TITLE
Scenes: Setting default_home_dashboard_path returns blank page and no controls

### LINKED ISSUE #89112: Dashboard scenes: Home page is blank when dashboardScene and default_home_dashboard_path is set on Grafana 11
### What happened?

We trying to upgrade to Grafana 11 with using dashboards scene.

Unfortunately home dashboard get's blank after enabling scenes.
image

Blank page fixes somehow during window resize so it's possible to get menu page on left

image

### What did you expect to happen?

We expect to see home dashboard, which is defined in configuration

### Did this work before?

We haven't used scenes before.

### How do we reproduce it?

1. Setup Grafana 11 with dashboardScene feature and default_home_dashboard_path is set to valid dashboard
2. If dashboard is invalid you will see notification and `[Object] Object` blank page

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: 11
OS: Windows
Browser: Chome and Edge

### Grafana platform?

Kubernetes

### Datasource(s)?

bitnami helm chart

### PR DESCRIPTION
**Pull request by ivanortegaalba**
**Issue**

Fixes #89112 

When setting a custom page, the parameters don't include the dashboard UID. Because of the condition to not re-render when moving between dashboards introduced here https://github.com/grafana/grafana/pull/86262 it was returning null and not recovering from that state.

**Solution**
Do not apply that logic when the route is the home page.

**EXTRA CHANGES**
I don't know the reason, but the controls were hidden for the home page. This doesn't apply to custom dashboards either. I removed that condition since it doesn't make sense.

### CODE REVIEW

**Comment by dprokop** [comment]
Ouch, thought it's not assigned so i can take it.... https://github.com/grafana/grafana/pull/89305

**Comment by dprokop** [comment]
Btw - i'm not sure this PR fixes the problem: 
image

**Comment by ivanortegaalba** [comment]
I'm solving the issue when the custom home page is valid path from `custom.ini`. You're solving the issue when it is not valid, that was broken too.

This is the comparison before and after applying my changes:
|Before|After|
|-|-|
Captura de pantalla 2024-06-17 a las 18 04 27|Captura de pantalla 2024-06-17 a las 18 03 39

Can you try to test the following?

1. Add to your custom.ini the following:
```
[dashboards]
default_home_dashboard_path = /<my_path>/grafana/data/annotation-filtering.json
```
2. Move from gdev-dashboards the dashboard annotation-filtering.json to the configured folder
3. Run the server and the client locally. With your fix, the result still a blank page if I'm not wrong.

**Comment by dprokop** [comment]
haha, you are right, we are fixing two different bugs - I think we have differently interpreted the issue and hence got two different bugs fixed :D

**Comment by ivanortegaalba** [comment]
I added some tests :) Everything should be ready to be reviewed now. I'm reviewing yours :D

File: public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export function DashboardSceneRenderer({ model }: SceneComponentProps<DashboardS
         >
           {scopes && <scopes.Component model={scopes} />}
           <NavToolbarActions dashboard={model} />
-          {!isHomePage && controls && hasControls && (
+          {controls && hasControls && (
```
Comment (dprokop) [inline public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx:60]: @ivanortegaalba - this is not correct, with this change the default home dashboard gets time controls which should not happen. I think we should simply rely on the home dashboard definition, for some reason the default home dashboard shows the controls. Seems like the `controls.hasControls()` may be invalid implementation?

**Review by dprokop — CHANGES_REQUESTED** [review]
It does fix the mentioned bug, but also introduces a regression to the default home page.

File: public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx (hunk 1 above)
Comment (ivanortegaalba) [reply public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx:60]: Fixed :) it was a bug with the URL sync. Now it should be fixed

**Comment by ivanortegaalba** [comment]
@dprokop the issue is solved now. Good catch
|Before|After|
|-|-|
Captura de pantalla 2024-06-18 a las 15 38 19|Captura de pantalla 2024-06-18 a las 15 38 26

**Review by dprokop — APPROVED** [review]
Works as expected. One last thing I see is spacing between the grid and the controls thats different on home dashboar vs regular dashboard. SO, for the Annotations filtering dashboard, when used as home dashboard the gap is bigger:

image



The same dashboard but not used as home:

image

**Comment by ivanortegaalba** [comment]
@dprokop good catch! I fixed it here https://github.com/grafana/grafana/pull/89432

**Pull request merged**